app/ui/legacy/src/main/java/com/fsck/k9/activity/setup/AccountSetupCheckSettings.java

```diff
@@ -251,13 +244,24 @@ public void run() {
                     }
 
                     chainInfo.append("Issuer: ").append(chain[i].getIssuerDN().toString()).append("\n");
-                    if (sha1 != null) {
-                        sha1.reset();
+                    String digestAlgorithms[] = getResources().getStringArray(R.array.message_digest_algorithms);
```

Comment:
There's no need to read the algorithm identifiers from resources. Please put them here instead.

Code style: Use `String[] name` instead of `String name[]`.

```suggestion
                    String[] digestAlgorithms = { "SHA-1", "SHA-256", "SHA-512" };
```